dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -763,6 +766,603 @@ def generate_TD_waveform(self, parameters_lal: Tuple) -> Dict[str, np.ndarray]:
+        harmonics).
+
+        Note:
+            - pol_m[m] contains contributions of the m modes *and* and the -m modes.
+              This is because the frequency domain (FD) modes have a positive frequency
+              part which transforms as exp(-1j * m * phase), while the negative
+              frequency part transforms as exp(+1j * m * phase). Typically, one of these
+              dominates [e.g., the (2,2) mode is dominated by the negative frequency
+              part and the (-2,2) mode is dominated by the positive frequency part]
+              such that the sum of (l,|m|) and (l,-|m|) modes transforms approximately as
+              exp(1j * |m| * phase), which is e.g. used for phase marginalization in
+              bilby/lalinference. However, this is not exact. In this method we account
+              for this effect, such that each contribution pol_m[m] transforms
+              *exactly* as exp(-1j * m * phase).
+            - Phase shifts contribute in two ways: Firstly via the spherical harmonics,
+              which we account for with the exp(-1j * m * phase) transformation.
+              Secondly, the phase determines how the PE spins transform to cartesian
+              spins, by rotating (sx,sy) by phase. This is *not* accounted for in this
+              function. Instead, the phase for computing the cartesian spins is fixed
+              to self.spin_conversion_phase (if not None). This effectively changes the
+              PE parameters {phi_jl, phi_12} to parameters {phi_jl_prime, phi_12_prime}.
+              For parameter estimation, a postprocessing operation can be applied to
+              account for this, {phi_jl_prime, phi_12_prime} -> {phi_jl, phi_12}.
+              See also documentation of __init__ method for more information on
+              self.spin_conversion_phase.
+
+        Differences to self.generate_hplus_hcross:
+        - We don't catch errors yet TODO
+        - We don't apply transforms yet TODO
+
+        Parameters
+        ----------
+        parameters: dict
+            Dictionary of parameters for the waveform.
+            For details see see self.generate_hplus_hcross.
+
+        Returns
+        -------
+        pol_m: dict
+            Dictionary with contributions to h_plus and h_cross, sorted by their
+            transformation behaviour under phase shifts:
+            {m: {"h_plus": hp_m, "h_cross": hc_m} for m in [-l_max,...,0,...,l_max]}
+            Each contribution h_m transforms as exp(-1j * m * phase) under phase shifts
+            (for fixed self.spin_conversion_phase, see above).
+        """
+        if not isinstance(parameters, dict):
+            raise ValueError("parameters should be a dictionary, but got", parameters)
+        elif not isinstance(list(parameters.values())[0], float):
+            raise ValueError("parameters dictionary must contain floats", parameters)
+
+        generator = gwsignal_get_waveform_generator(self.approximant_str)
```

Comment:
What is the runtime for this function? Does it make sense to save the generator as an attribute to be able to access it faster?

Reply:
It is quite fast, just calls the generator class of the approximant. Storing the approximant generator as an attribute produces a pickle error I couldn't manage to solve, therefore I think it is fine leaving calling this function when needed